### ISSUE TITLE
Match digits with roman numbers

### ISSUE THREAD

**Issue by ntncmch** [epicentre-msf/hmatch#11]
Often level names with a digit can be spelled with a roman number:

Ex: Paris 18 vs Paris XVIII

It would be nice to add an option to automatically convert and test for potential match.

**Comment by patrickbarks** [comment]
Good idea. I've added a `convert_roman` argument to the `string_std()` function to convert Roman numerals to Arabic (just 1-49 for now), and added a `...` argument to all `hmatch_` functions to enable passing additional arguments to whatever function is passed to `std_fn`. E.g.

``` r
raw <- data.frame(
  adm1 = "Île-de-France",
  adm2 = "Paris",
  adm3 = "Paris XVIII"
)

ref <- data.frame(
  adm1 = "Île-de-France",
  adm2 = "Paris",
  adm3 = "Paris 18"
)

# string_std now has option to convert roman numerals to arabic
hmatch::string_std("Paris XVIII")
#> [1] "paris_xviii"
hmatch::string_std("Paris XVIII", convert_roman = TRUE)
#> [1] "paris_18"

# no conversion by default, so no match
hmatch::hmatch_complete(raw, ref)
#>            adm1  adm2        adm3 ref_adm1 ref_adm2 ref_adm3
#> 1 Île-de-France Paris Paris XVIII     <NA>     <NA>     <NA>

# pass convert_roman = TRUE to str_std to obtain match
hmatch::hmatch_complete(raw, ref, convert_roman = TRUE)
#>            adm1  adm2        adm3      ref_adm1 ref_adm2 ref_adm3
#> 1 Île-de-France Paris Paris XVIII Île-de-France    Paris Paris 18
```

<sup>Created on 2020-05-26 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by patrickbarks** [comment]
closed previously by 04b3d4a
